src/misc/mntent.c

```diff
@@ -20,6 +20,34 @@ int endmntent(FILE *f)
 	return 1;
 }
 
+static inline int decode1(char** in_buf, char** out_buf,  const char* from, const char to) {
+	if(strncmp(from, *in_buf, strlen(from)) == 0) {
+		*(*out_buf)++ = to;
+		*in_buf += strlen(from);
+		return 1;
+	}
+	return 0;
+}
+
+static inline char* decode(char* buf) {
+	char* read_cursor = buf;
+	char* write_cursor = buf;
+	while(read_cursor && *read_cursor) {
+			// space
+			decode1(&read_cursor, &write_cursor, "\\040", '\040')
```

Comment:
i'd first check if the character is '\\' and only call the costly decode func in that case. the different values could be stored in a small table instead of having a branch for each one.

Reply:
Eh, I guess I'll have to run a benchmark and check with Godbolt.

My prior is that this kind of micro-optimization doesn't make a difference, but we can check.

Reply:
the difference is 2 strlen + 1 strncmp call per character and per case you hardcoded even when the current char isn't  '\\'. additionally you can then compare against the strings with the leading backslash removed, saving one byte per string in .text. musl's codestyle is dense, efficient, and optimized for size.

Reply:
Ok, I played a bit with Godbolt, almost everything gets inlined here.

Reply:
Sorry, I don't understand: there's no strlen call at all in the binary. There are some strlen calls in the source, but they all get evaluated at compile time and replaced with constants in the binary.

When you say optimised for size, are you talking about size of the source code or size of the binary?

Reply:
> but they all get evaluated at compile time and replaced with constants in the binary.

just because the compiler you test it with does doesn't mean everybody else's does. people use different compiler versions (way back to gcc 3.4.6) and optimization flags

> When you say optimised for size, are you talking about size of the source code or size of the binary?

generally binary size, but that is usually a side effect of small code size and being clever

> EDIT: At least in Godbolt, there's nothing in .text either, it all gets inlined.

iirc godbolt doesn't show read only data saved in .text, like the string literals you use, but they *are* in the object files and binaries using them

Reply:
Your argument about other compilers is sound.  I'll look into changing the code as you request.

Btw, Godbolt does show the .text segment: I can see those strings in there, when I ask it to compile with `-O0`.